File: Marathon Solutions/September Marathon/SCITST.java
import java.util.*;

 class solution {
    int[] visited;
    void explore(int u, HashMap<Integer, ArrayList<Integer>> d, ArrayList<Integer> l,int[] visited) {

        visited[u] = 1;
        l.add(u);
        for(int i: d.get(u)) {
            if(visited[i] == 0) {
                explore(i, d, l, visited);
            }
        }
    }
    ArrayList<ArrayList<Integer>> connect(HashMap<Integer, ArrayList<Integer>> d, int[]visited) {
        ArrayList<Integer> l = new ArrayList<>();
        ArrayList<ArrayList<Integer>> l1 =new ArrayList<>();
        for(int i = 0; i < d.size(); i++) {
            if(visited[i] == 0) {
                explore(i, d, l, visited);
                l1.add(l);
                l = new ArrayList<>();
            }
        }
        return (l1);
    }
    public void solve() {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        visited = new int[n];
        int m = in.nextInt();
        HashMap<Integer, ArrayList<Integer>> d = new HashMap<>();
        for(int i = 0; i < n; i++) {
            d.put(i, new ArrayList<>());
        }

        for(int i = 0; i < m; i++) {
            int u = in.nextInt();
            int v = in.nextInt();
            d.get(u).add(v);
            d.get(v).add(u);
        }
        ArrayList<ArrayList<Integer>> l=connect(d,visited);
        int s=(n*(n-1))/2;

        for (ArrayList<Integer> integers : l) {
            n = integers.size();
            s -= (n * (n - 1)) / 2;
        }
        System.out.println(s);
    }

    public static void main(String[] args) {
        (new solution()).solve();
    }
}

File: Marathon Solutions/September Marathon/SCITST.py
#python3

def explore(u,d,l,visited):
  
  visited[u]=1
  l.append(u)
  for i in d[u]:
    if visited[i]==0:
      explore(i,d,l,visited)
def connect(d,visited):
  l=[]
  l1=[]
  for i in d:
    if visited[i]==0:   
      explore(i,d,l,visited)
      l1.append(l)
      l=[]
      
  return(l1)
  
n,m=map(int,input().split())
d={}
for i in range(n):
  d[i]=[]

for i in range(m):
  u,v=map(int,input().split())
  d[u].append(v)
  d[v].append(u)
visited=[0]*(n)

l=connect(d,visited)
n1=len(l)
s=(n*(n-1))//2

for i in range(n1):
    n=len(l[i])
    s-=(n*(n-1))//2
print(s)


